gooddata-java-model/src/main/java/com/gooddata/sdk/model/hierarchicalconfig/SourceType.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.model.hierarchicalconfig;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.gooddata.sdk.common.GoodDataException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Source type defines source of setting, its api uri in case it is manageable and source hierarchy of setting.
+ * For example, you can configure a whole domain and overwrite that configuration item for the project setting.
+ */
+public enum SourceType {
+
+    @JsonProperty("catalog")
+    CATALOG("catalog", null),
+    @JsonProperty("client")
+    CLIENT("client", "/gdc/domains/{domainName}/dataproducts/{dataProductId}/clients/{clientId}/config"),
+    @JsonProperty("data_product")
+    DATA_PRODUCT("data_product", "/gdc/domains/{domainName}/dataproducts/{dataProductId}/config"),
+    @JsonProperty("segment")
+    SEGMENT("segment", "/gdc/domains/{domainName}/dataproducts/{dataProductId}/segments/{segmentId}/config"),
+    @JsonProperty("datawarehouse")
+    DATAWAREHOUSE("datawarehouse", null),
+    @JsonProperty("domain")
+    DOMAIN("domain", "/gdc/domains/{domainName}/config"),
+    @JsonProperty("project")
+    PROJECT("project", "/gdc/projects/{pid}/config"),
+    @JsonProperty("project_group")
+    PROJECT_GROUP("project_group", "/gdc/projectGroups/{projectGroup}/config"),
+    @JsonProperty("user")
+    USER("user", null);
+
+    private static final Map<String, SourceType> lookup = new HashMap<>();
+
+    static {
+        DOMAIN.possibleValues = Arrays.asList(CATALOG, DOMAIN);
+        PROJECT_GROUP.possibleValues = Arrays.asList(CATALOG, DOMAIN, PROJECT_GROUP);
+        PROJECT.possibleValues = Arrays.asList(CATALOG, DOMAIN, PROJECT_GROUP, PROJECT);
```

Comment:
Shouldn't there be also `DATA_PRODUCT, SEGMENT, CLIENT`?

Reply:
After the agreement, the `possibleValues` member was removed.